sample-app/src/main/java/com/sinch/sample/numbers/NumbersSampleFlow.java

```diff
@@ -0,0 +1,392 @@
+
+    echo(
+        "SUCCESS: The number "
+            + phoneNumber
+            + " has been found as an active number. Cost is "
+            + activeNumber.getMoney().getAmount()
+            + " "
+            + activeNumber.getMoney().getCurrencyCode()
+            + "\n");
+  }
+
+  void verifyActiveNumberByPagination(
+      int step, SinchClient sinchClient, String phoneNumber, String regionCode) {
+
+    echoStep(
+        step,
+        "Check the number is part of our active numbers - Method 2: List all active numbers and"
+            + " navigate across pages");
+
+    // 1. Build the request data
+    var listActiveNumbersRequestData =
+        ActiveNumberListRequestParameters.builder()
+            .setRegionCode(regionCode)
+            .setType(NumberType.LOCAL)
+            .setPageSize(2) // Set a low number to demonstrate the pagination later on
+            .build();
+
+    // 2. Request for active number using the built-in SDK method
+    ActiveNumberListResponse activeNumbersListResponse =
+        sinchClient.numbers().active().list(listActiveNumbersRequestData);
+
+    // 3. To check if the phone number is part of the active numbers: check the page
+    // content and if not present, check the next page until we find the phone number, or we reach
+    // the end of the list
+    Optional<ActiveNumber> found;
+    do {
+      found =
+          activeNumbersListResponse.getContent().stream()
+              .filter(
+                  f ->
+                      phoneNumber.equals(f.getPhoneNumber())
+                          && regionCode.equals(f.getRegionCode()))
+              .findFirst();
+      if (found.isEmpty() && activeNumbersListResponse.hasNextPage()) {
+        activeNumbersListResponse = activeNumbersListResponse.nextPage();
+      } else {
+        activeNumbersListResponse = null;
+      }
+    } while (found.isEmpty() && null != activeNumbersListResponse);
+
+    echo(
+        (found.isPresent() ? "SUCCESS:" : "FAILED:")
+            + " The number "
+            + phoneNumber
+            + " is "
+            + (found.isPresent() ? "" : "NOT ")
+            + "listed in the active numbers.\n");
+  }
+
+  void verifyActiveNumberByAutoPagination(
+      int step, SinchClient sinchClient, String phoneNumber, String regionCode) {
+
+    echoStep(
+        step,
+        "Check the number is part of our active numbers - Method 3: List all active numbers and"
+            + " navigate across pages using auto pagination SDK feature");
+
+    // 1. Build the request data
+    var listActiveNumbersRequestData =
+        ActiveNumberListRequestParameters.builder()
+            .setRegionCode(regionCode)
+            .setType(NumberType.LOCAL)
+            .setPageSize(2) // Set a low number to demonstrate the pagination later on
+            .build();
+
+    // 2. Request for active number using the built-in SDK method
+    ActiveNumberListResponse activeNumbersListResponse =
```

Comment:
The steps 1 and 2 are duplicated from `verifyActiveNumberByPagination`. I understand we need to "reset" the activeNumbersListResponse in order to run the iterator from the first elements returned by the server.
But in this example, it's maybe clearer to have a duplication rather to try to factorise this part

Reply:
Will synch; need to clarify what you mean